piptools/writer.py

```diff
@@ -104,12 +125,17 @@ def write_flags(self):
     def _iter_lines(
         self,
         results,
-        unsafe_requirements,
-        reverse_dependencies,
-        primary_packages,
-        markers,
-        hashes,
+        unsafe_requirements=[],
```

Comment:
Why not `unsafe_requirements=None`?

Reply:
`None` works for me -- updated.

(I think my intuition here is to not make something a maybe/optional type if it doesn't have to be -- it's easier to write correct code against "zero or more items in a list" than against "zero or more items in a list, or nothing." But I don't feel strongly about it.)

Reply:
I'm just saying that mutable default values should be avoided (unless it's done on purpose for some reason). BTW, the rest of defaults of arguments are still mutable.

Reply:
Oh, I see. I updated the code to set all default values within the function.

Personally I think mutable default values are fine, because it should be the baseline assumption that a function does not modify its inputs anyway. If a function does modify its inputs then it shouldn't have mutable defaults, but it should also clearly express what it does in the signature so callers know what to expect. But again, don't feel strongly about it. :)